codec-http/src/main/java/io/netty/handler/codec/http/QueryStringEncoder.java

```diff
@@ -69,10 +72,19 @@ public void addParam(String name, String value) {
             uriBuilder.append('?');
             hasParams = true;
         }
-        appendComponent(name, charsetName, uriBuilder);
+
+        encodeComponent(name);
         if (value != null) {
             uriBuilder.append('=');
-            appendComponent(value, charsetName, uriBuilder);
+            encodeComponent(value);
+        }
+    }
+
+    private void encodeComponent(CharSequence s) {
+        if (CharsetUtil.UTF_8.equals(charset)) {
```

Comment:
imho we can do the equals check once in the constructor, store the result in an field and so reduce the overhead even more..

Reply:
so why not just use `==` ?The Charset instance was cache returned and I think there should be only one instance pre charset in jvm...

Reply:
IMHO it is not guaranteed that there will only be one instance... I would rather be safe then sorry ;)

Reply:
all right,I will fix this